pkg/controllers/externalsecret/externalsecret_controller.go

```diff
@@ -320,36 +342,47 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	}, nil
 }
 
-func createOrUpdate(ctx context.Context, c client.Client, obj client.Object, f func() error, fieldOwner string) error {
+func isOrphanSecret(externalSecret esv1beta1.ExternalSecret, newSecret *v1.Secret) bool {
+	secretRef := externalSecret.Status.CreatedSecretReference
+	if externalSecret.Spec.Target.CreationPolicy != esv1beta1.CreatePolicyOwner ||
```

Comment:
@gusfcarvalho RE https://github.com/external-secrets/external-secrets/pull/1389#issuecomment-1527209121

> (possibly dumb) question: why are creationPolicy=Owner secrets not affected? They get cleaned up only once the ExternalSecret gets deleted. So if the spec.target.name changes, it'll create a new secret and leave the old one as-is or am i mistaken?

Context for this comment :point_up_2:  is this line.

 `isOrphanSecret()` is only called if `externalSecret.Spec.Target.CreationPolicy` is `Owner` which makes this check superfluous i think :thinking:

Reply:
I can remove the redundancies as well